types/ramda/test/map-tests.ts

```diff
@@ -36,9 +36,20 @@ import * as R from 'ramda';
     b: string;
   }
 
+  interface C {
+    b: number;
+    c: string;
+  }
+
   R.map<A, A>(R.inc, { a: 1, b: 2 });
   R.map<A, B>(R.toString, { a: 1, b: 2 });
 
   R.map<A, A>(R.inc)({ a: 1, b: 2 });
   R.map<A, B>(R.toString)({ a: 1, b: 2 });
+
+  type KeyOfUnion<T> = T extends infer U ? keyof U : never;
+
+  R.map<A | C, Record<KeyOfUnion<A | C>, void>>(value => {
+    if (typeof value === 'string') console.log(value.toUpperCase());
+  }, { a: 1, b: 2 });
```

Comment:
Can you add some [`// $ExpectType <type>`](https://github.com/microsoft/dtslint#write-tests) comments?

As-is, this test doesn't actually test a whole lot, especially about the output.

I'd hope for something like:

```suggestion
  // $ExpectType Record<KeyOfUnion<A | C>, void>
  R.map<A | C, Record<KeyOfUnion<A | C>, void>>(
    // $ExpectType (value: A | C) => void
    value => {
      if (typeof value === 'string') console.log(value.toUpperCase());
    },
    { a: 1, b: 2 },
  );
```

You'll probably have to play with it a bit — those `$ExpectType` assertions probably won't work as-is — but you get the idea.